tt_metal/impl/dispatch/command_queue.cpp

```diff
@@ -1144,47 +1169,80 @@ void EnqueueProgramCommand::assemble_device_commands() {
             }
             i++;
         }
+        uint32_t go_signal_count = 0;
         for (auto& go_signal : cached_program_command_sequence.go_signals) {
             go_signal->kernel_config.dispatch_core_x = this->dispatch_core.x;
             go_signal->kernel_config.dispatch_core_y = this->dispatch_core.y;
+            if (go_signal_count < program.tensix_go_signal_count_) {
+                go_signal->kernel_config.kernel_config_base = tensix_l1_kernel_config_base;
+            } else {
+                go_signal->kernel_config.kernel_config_base = eth_l1_kernel_config_base;
+            }
+            go_signal_count++;
         }
     }
 }
 
 void EnqueueProgramCommand::process() {
-    // Calculate all commands size and determine how many fetch q entries to use
 
-    // Preamble, some waits and stalls
-    // can be written directly to the issue queue
+    bool is_cached = true;
     if (not program.is_finalized()) {
-        // TODO: only build kernel groups here, remove on-the-fly state validation and validate here
-        program.finalize_rt_args();
+        program.finalize();
+        is_cached = false;
+    }
+
+    const std::pair<ConfigBufferSync, std::vector<ConfigBufferEntry>&> reservation =
+        this->manager.get_config_buffer_mgr().reserve(program.program_config_sizes_);
+    bool stall_first = reservation.first.need_sync;
+    // Note: since present implementation always stalls, we always free up to "now"
+    this->manager.get_config_buffer_mgr().free(reservation.first.sync_count);
+    this->manager.get_config_buffer_mgr().alloc(this->expected_num_workers_completed +
```

Comment:
Is this why we don't set offsets for our `write_packed`?

Reply:
Not sure what you mean.  The write_packed commands are generated once and contain offsets from a base address.  The base address is updated for each enqueue program w/ set_write_offset.

Reply:
I saw that `add_dispatch_set_write_offsets` wasn't being called in `command_queue.cpp`. Maybe I'm missing something.

Reply:
It's called from assemble preamble commands

Reply:
Ah shoot, missed it ... the file was minimized out. Thanks!